mtags/src/main/scala/scala/meta/internal/pc/ReferenceProvider.scala

```diff
@@ -0,0 +1,18 @@
+package scala.meta.internal.pc
+
+import org.eclipse.{lsp4j => l}
+
+trait ReferenceProvider {
+  val text: Array[Char]
+  val range: l.Range
+  def result(): List[Either[Reference, Definition]]
+}
+
+case class Definition(
+    range: l.Range,
+    rhs: String,
+    rangeOffsets: (Int, Int),
```

Comment:
```suggestion
    rangeOffsets: l.Position
```
?

Reply:
It's the start and end offset, so it marks two positions really. I am not convinced this it the best way to do this, but I really just need positions marked as `Int` since it's used to slice the whole text. I guess I could return `l.Range` and have the text split into lines, but it feels a bit error prone without a method that can deal with escaped symbols well.

Reply:
Maybe let's do an additional case class the? 
```scala
case class RangeOffset(start: Int, end: Int)
```